ublox/src/ubx_packets/packets.rs

```diff
@@ -1020,6 +1159,163 @@ impl<T: FloatCore + FromPrimitive + ToPrimitive> ScaleBack<T> {
     }
 }
 
+/// Navigation Engine Expert Settings
+#[ubx_packet_recv_send]
+#[ubx(
+    class = 0x06,
+    id = 0x23,
+    fixed_payload_len = 40,
+    flags = "default_for_builder"
+)]
+struct CfgNavX5 {
+    /// Only version 2 supported
+    version: u16,
+
+    /// Only the masked parameters will be applied
+    #[ubx(map_type = CfgNavX5Params1)]
+    mask1: u16,
+
+    #[ubx(map_type = CfgNavX5Params2)]
+    mask2: u32,
+
+    /// Reserved
+    reserved1: [u8; 2],
+
+    /// Minimum number of satellites for navigation
+    min_svs: u8,
+
+    ///Maximum number of satellites for navigation
+    max_svs: u8,
+
+    /// Minimum satellite signal level for navigation
+    min_cno_dbhz: u8,
+
+    /// Reserved
+    reserved2: u8,
+
+    /// initial fix must be 3D
+    ini_fix_3d: u8,
+
+    /// Reserved
+    reserved3: [u8; 2],
+
+    /// issue acknowledgements for assistance message input
+    ack_aiding: u8,
```

Comment:
Just for my own understanding, if you don't mind my asking, do you know what version of the protocol you're using? I haven't yet decided what the best way to handle version discrepancies is, but until then I'm trying to get a sense of what versions people are using.

Reply:
We use a number of different models of GPS with protocol versions 15, 17, and 22. Handling multiple versions of the same message is going to be messy, I'm not sure what the best approach is. So far we haven't had to deal with it.